client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -31,14 +34,19 @@
 
 public class CachedSchemaRegistryClient implements SchemaRegistryClient {
 
-  private final String baseUrl;
+  private UrlList baseUrls;
   private final int identityMapCapacity;
   private final Map<String, Map<Schema, Integer>> schemaCache;
   private final Map<Integer, Schema> idCache;
   private final Map<String, Map<Schema, Integer>> versionCache;
 
-  public CachedSchemaRegistryClient(String baseUrl, int identityMapCapacity) {
-    this.baseUrl = baseUrl;
+  public static CachedSchemaRegistryClient newInstance(String baseUrlConfig, int identityMapCapacity) {
+    List<String> baseUrls = Arrays.asList(baseUrlConfig.split(","));
```

Comment:
Validation? We probably want to make sure the list is at least non-empty. Although this may have just been missing before too.

Actually, you could probably just get this for free out of the Config class. There's list-of-string parsing in confluent-common's AbstractConfig class, which is what AbstractKafkaAvroSerDeConfig uses. I think just changing the type of that option to a list (and handling the subsequent fallout of having to call `config.getList()` instead of `config.getString()`) would also work.